components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImplTest.java

```diff
@@ -0,0 +1,360 @@
+        Mockito.when(serviceReferenceHolder.getRealmService()).thenReturn(realmService);
+        TenantManager tenantManager = Mockito.mock(TenantManager.class);
+        Mockito.when(realmService.getTenantManager()).thenReturn(tenantManager);
+        Mockito.when(tenantManager.getTenantId(TENANT_DOMAIN)).thenReturn(TENANT_ID);
+
+        PowerMockito.mockStatic(MultitenantUtils.class);
+        PowerMockito.doReturn(TENANT_DOMAIN).when(MultitenantUtils.class, "getTenantDomain", ADMIN_USERNAME);
+
+        PowerMockito.mockStatic(ApiMgtDAO.class);
+        apiMgtDAO = Mockito.mock(ApiMgtDAO.class);
+        PowerMockito.doReturn(apiMgtDAO).when(ApiMgtDAO.class, "getInstance");
+
+        endpointRegistry = new EndpointRegistryImpl(ADMIN_USERNAME);
+    }
+
+    @Test
+    public void addEndpointRegistry() throws APIManagementException {
+        EndpointRegistryInfo endpointRegistryInfo = new EndpointRegistryInfo();
+        endpointRegistryInfo.setName("Endpoint Registry 1");
+        endpointRegistryInfo.setMode("ReadOnly");
+        endpointRegistryInfo.setType("wso2");
+        endpointRegistryInfo.setOwner(ADMIN_USERNAME);
+        endpointRegistryInfo.setUuid("abc1");
+
+        Mockito.when(apiMgtDAO.isEndpointRegistryNameExists(endpointRegistryInfo.getName(), TENANT_ID))
+                .thenReturn(false);
+        Mockito.when(apiMgtDAO.addEndpointRegistry(endpointRegistryInfo, TENANT_ID))
+                .thenReturn(endpointRegistryInfo.getUuid());
+
+        String registryUUID = endpointRegistry.addEndpointRegistry(endpointRegistryInfo);
+
+        Assert.assertEquals(endpointRegistryInfo.getUuid(), registryUUID);
+    }
+
+    @Test(expected = APIMgtResourceAlreadyExistsException.class)
+    public void addEndpointRegistry_existingEntryName() throws APIManagementException {
+        EndpointRegistryInfo endpointRegistryInfo = new EndpointRegistryInfo();
+        endpointRegistryInfo.setName("Endpoint Registry 1");
+        endpointRegistryInfo.setMode("ReadOnly");
+        endpointRegistryInfo.setType("wso2");
+        endpointRegistryInfo.setOwner(ADMIN_USERNAME);
+        endpointRegistryInfo.setUuid("abc1");
+
+        Mockito.when(apiMgtDAO.isEndpointRegistryNameExists(endpointRegistryInfo.getName(), TENANT_ID))
+                .thenReturn(true);
+
+        endpointRegistry.addEndpointRegistry(endpointRegistryInfo);
+    }
+
+    @Test
+    public void updateEndpointRegistry() throws APIManagementException {
+        EndpointRegistryInfo endpointRegistryInfo = new EndpointRegistryInfo();
+        endpointRegistryInfo.setName("Endpoint Registry 1");
+        endpointRegistryInfo.setMode("ReadOnly");
+        endpointRegistryInfo.setType("wso2");
+        endpointRegistryInfo.setOwner(ADMIN_USERNAME);
+        endpointRegistryInfo.setUuid("abc1");
+
+        endpointRegistry.updateEndpointRegistry(endpointRegistryInfo.getUuid(), endpointRegistryInfo.getName(),
+                endpointRegistryInfo);
```

Comment:
shall we verify whether an expected method was called or something since we cannot assert?

Reply:
updated in https://github.com/wso2/carbon-apimgt/pull/8534/commits/8aad6c2009f4840cedef91797b02feb0a640fa4e